mina-daemon/templates/deployment.yaml

```diff
@@ -470,6 +474,24 @@ spec:
               mountPath: /usr/bin/gcloud
               subPath: gcloud-impersonator.sh
             {{- end }}
+
+        {{- if .Values.node.exposeGraphql }}
+        - name: graphql-public-proxy
+          image: "{{ .Values.graphqlPublicProxy.image.repository }}:{{ .Values.graphqlPublicProxy.image.tag }}"
+          imagePullPolicy: {{ .Values.graphqlPublicProxy.image.pullPolicy }}
+          env:
+            - name: MINA_GRAPHQL_PORT
+              value: {{ .Values.node.ports.graphql | quote }}
+            - name: EXTERNAL_PORT
+              value: {{ .Values.node.ports.proxy | quote }}
+          ports:
+            - name: proxy
+              protocol: TCP
+              containerPort: {{ .Values.node.ports.proxy }}
+          resources:
+            {{- toYaml .Values.graphqlPublicProxy.resources | nindent 12 }}
+        {{- end -}}
+
```

Comment:
Public proxy runs as a sidecar container. It listen to a specific port (`3000`) and forward the traffic to the normal graphql port. It also obfuscates some sensitive fields and functionalities (e.g. mutations)